### ISSUE TITLE
Linux & gcc bug

### ISSUE THREAD

**Issue by iryont** [ThePhD/infoware#42]
In system.hpp during compilation we get this error:

```bash
/usr/local/include/infoware/system.hpp:32:4: error: expected identifier before numeric constant
   32 |    linux,
      |    ^~~~~
/usr/local/include/infoware/system.hpp:32:4: error: expected '}' before numeric constant
```

It probably happens because on Linux gcc defines "linux" pre-processor definition, so the easiest way for me to "fix" it was to #undef linux before.

**Comment by nabijaczleweli** [comment]
Can you try with 627a7355a1e63a519580b3c1604ce3e8e2517783?

**Comment by iryont** [comment]
Indeed, that will do the trick. I did the same to compile infoware :)

**Comment by nabijaczleweli** [comment]
Released in [`v0.5.2`](https://github.com/ThePhD/infoware/releases/tag/v0.5.2); thanks!
